pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -95,6 +98,22 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 					}
 					// TODO: should we skip the node's own IP like we do in listener?
 					addressList = append(addressList, instance.Endpoint.Address)
+					if pod, ok := instance.Endpoint.Labels[StatefulSetPodLabel]; ok {
+						address := []string{instance.Endpoint.Address}
+						// Follow k8s naming convention of "pod.svcHostname" i,e. "pod.svc.svcns.dnsdomain".
+						host := pod + "." + string(svc.Hostname)
```

Comment:
I think this may add dns entries which should be invalid.  Say there are two services selecting the statefulset pods.

Reply:
Is that possible and how do we handle that?

Reply:
Yes, possible. In k8s, it is common to set two services for a statefulset, one if of clusterip and the other is headless. I will look into how to prevent it.

Reply:
But this we are only handling for headless - are you saying clusterIP DNS entries may be incorrect? Any way will wait for your fix.

Reply:
Just noticed the `svc.Resolution == model.Passthrough`, means the service can only be headless service. 

So here the problem will only occur when more than one headless service selecting the statefulset pods in very rare case.